### ISSUE TITLE
Add psr/http-message: ^2.0 support

### ISSUE THREAD

**Issue by babenkoivan** [elastic/elasticsearch-php#1316]
### Summary of problem or feature request

The latest Laravel version comes with `psr/http-message` locked to version 2. `elasticsearch-php` requires `psr/http-message: ^1.0`, which makes it impossible to install the latest `elasticsearch-php` version without downgrading `psr/http-message`.

```
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Root composer.json requires elasticsearch/elasticsearch v8.7.1 -> satisfiable by elasticsearch/elasticsearch[v8.7.1].
    - elasticsearch/elasticsearch v8.7.1 requires psr/http-message ^1.0 -> found psr/http-message[1.0, 1.0.1, 1.1] but the package is fixed to 2.0 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.

Use the option --with-all-dependencies (-W) to allow upgrades, downgrades and removals for packages currently locked to specific versions.
```

### Steps to reproduce

```bash
composer create-project laravel/laravel test
cd test
composer require elasticsearch/elasticsearch:v8.7.1
```

### System details

- Operating System: macOS Ventura
- PHP Version: 8.2.3
- ES-PHP client version: 8.7.1
- Elasticsearch version: 8

**Comment by gjuric** [comment]
I've already created a PR for this https://github.com/elastic/elasticsearch-php/pull/1313 almost a month ago but didn't get a reaction.

**Comment by ezimuel** [comment]
@gjuric and @babenkoivan this is fixed now in main. I'll release 8.8.0 soon.

**Comment by gjuric** [comment]
Hi @ezimuel I see that you've released v8.8.0 and it states that this should be included in this release in the notes but this is not the case, it looks like some commits are missing and v8.8.0 still requires only v1 of psr/http-message.

**Comment by ezimuel** [comment]
@gjuric thanks for the notice, something went wrong during the release process. I just published a patch version [v8.8.1](https://github.com/elastic/elasticsearch-php/releases/tag/v8.8.1) containing #1313. Sorry for the inconvenience.
